Doc/library/doctest.rst

```diff
@@ -277,13 +277,34 @@ Which Docstrings Are Examined?
 The module docstring, and all function, class and method docstrings are
 searched.  Objects imported into the module are not searched.
 
-In addition, if ``M.__test__`` exists and "is true", it must be a dict, and each
+In addition, there are cases when you want tests to be part of a module but not part
+of the help text, which requires that the tests not be included in the docstring.
+Doctest looks for a module-level variable called ``__test__`` and uses it to locate other
+tests. If ``M.__test__`` exists and is ``True``, it must be a dict, and each
 entry maps a (string) name to a function object, class object, or string.
 Function and class object docstrings found from ``M.__test__`` are searched, and
 strings are treated as if they were docstrings.  In output, a key ``K`` in
-``M.__test__`` appears with name ::
+``M.__test__`` appears with name :samp:`{name of M}.__test__.K`.
```

Comment:
One last thing I'm a little confused on reading this—how is `name of M` different from just `M`, the placeholder for the name of the module? Shouldn't this just be _M_, just like everywhere else? Same as `K` is the placeholder for the key name? Is there something I'm missing here, or should this in fact read:

```suggestion
``M.__test__`` appears with name ``M.__test__.K``.
```

(Unless we want to do `` :samp:`{M}.__test__.{K}` ``, which is more semantically correct and perhaps a bit tad clearer for readers, but probably not enough to be worth the extra syntax everywhere?)